wiki/python/DEVELOPER_GUIDE.md

```diff
@@ -47,31 +47,110 @@ The examples in the documentation are checked through doctest. To skip the check
 Examples subject to numerical imprecision, or that can't be reproduced consistently should be skipped.
 
 ## Testing and Unit Testing
-We use [https://docs.pytest.org/en/latest/]() for writing and running tests. To see existing examples, refer to any of the `test_*.py` files in the folder `cuml/tests`.
-
-Some tests are run against inputs generated with [hypothesis](https://hypothesis.works/). See the `cuml/testing/strategies.py` module for custom strategies that can be used to test cuml estimators with diverse inputs. For example, use the `regression_datasets()` strategy to test random regression problems.
-
-When using hypothesis for testing, you must include at least one explicit example using the `@example` decorator alongside any `@given` strategies. This ensures that:
-1. Every test has at least one deterministic test case that always runs
-2. Critical edge cases are documented and tested consistently
-3. Test failures can be reproduced reliably
-
-Note: While the explicit examples will always run in CI, the hypothesis-generated test cases (from `@given` strategies) only run during nightly testing by default. This ensures fast CI runs while still maintaining thorough testing coverage.
-
-Example of a valid hypothesis test:
-```python
-@example(dtype=np.float32, sparse_input=False)  # baseline case, runs as part of PR CI
-@example(dtype=np.float64, sparse_input=True)   # edge case, runs as part of PR CI
-@given(
-    dtype=st.sampled_from((np.float32, np.float64)),
-    sparse_input=st.booleans()
-)  # strategy-based cases, only runs during nightly tests
-def test_my_estimator(dtype, sparse_input):
-    # Test implementation
-    pass
+We use [pytest](https://docs.pytest.org/en/latest/) for writing and running tests. To see existing examples, refer to any of the `test_*.py` files in the folder `cuml/tests`.
+
+### Test Organization
+- Keep all tests for a single estimator in one file, with exceptions for:
+  - Performance testing/benchmarking
+  - Generic estimator checks (e.g., `test_base.py`)
+- Use small, focused datasets for correctness testing
+- Only parametrize scale when it triggers alternate code paths
```

Comment:
```suggestion
- Only parametrize dataset size when it triggers alternate code paths
```
I find this easier to understand, if "scale" did refer to dataset size, if not that I'm lost as to what "scale" means here